eclair/src/diff.rs

```diff
@@ -1,17 +1,120 @@
-use std::collections::{HashMap, HashSet};
+use std::collections::{BTreeMap, BTreeSet};
+use std::fs::File;
+use std::io::prelude::*;
 
+use askama::Template;
 use chrono::Duration;
 use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
-use plotly::common::{Mode, Title};
-use plotly::layout::{Axis, Layout};
-use plotly::{Plot, Scatter};
+use itertools::EitherOrBoth;
+use itertools::Itertools;
 
 use crate::summary::Summary;
 use crate::summary_item::{ItemId, ItemQualifier};
+use std::path::PathBuf;
 
-fn extract_time_and_data(summary: &Summary) -> (Vec<String>, HashMap<&ItemId, usize>) {
+type Merged = Vec<EitherOrBoth<(usize, i64), (usize, i64)>>;
+
+fn merge(t1: &[i64], t2: &[i64]) -> Merged {
+    t1.iter()
+        .copied()
+        .enumerate()
+        .merge_join_by(t2.iter().copied().enumerate(), |i, j| i.1.cmp(&j.1))
+        .collect()
+}
+
+type Padded = Vec<[Option<f32>; 2]>;
```

Comment:
You could easily use an `ArrayVec` here and avoid allocation penalties.

Reply:
Could you elaborate?

Reply:
You should literally be able to `s/Vec/ArrayVec/` here and it Just Work® -- and then you save yourself a heap allocation at this level. :)­